src/global/reco/reco.cc

```diff
@@ -62,22 +62,28 @@ void InitPlugin(JApplication *app) {
     app->Add(new JChainFactoryGeneratorT<InclusiveKinematicsSigma_factory>(
             {"MCParticles", "ReconstructedChargedParticles", "ReconstructedChargedParticleAssociations"}, "InclusiveKinematicsSigma"));
 
-    app->Add(new JChainFactoryGeneratorT<GeneratedJets_factory>(
-            {"MCParticles"}, "GeneratedJets"));
-
-    app->Add(new JChainFactoryGeneratorT<ReconstructedJets_factory>(
-            {"ReconstructedParticles"}, "ReconstructedJets"));
-
     app->Add(new JChainFactoryGeneratorT<ReconstructedElectrons_factory>(
-        {"MCParticles", "ReconstructedChargedParticles", "ReconstructedChargedParticleAssociations",
-        "EcalBarrelScFiClusterAssociations",
-        "EcalEndcapNClusterAssociations",
-        "EcalEndcapPClusterAssociations",
-        "EcalEndcapPInsertClusterAssociations",
-        "EcalLumiSpecClusterAssociations",
-        },
-        "ReconstructedElectrons"
+         {"MCParticles", "ReconstructedChargedParticles", "ReconstructedChargedParticleAssociations",
+         "EcalBarrelScFiClusterAssociations",
+         "EcalEndcapNClusterAssociations",
+         "EcalEndcapPClusterAssociations",
+         "EcalEndcapPInsertClusterAssociations",
+         "EcalLumiSpecClusterAssociations",
+         },
+         "ReconstructedElectrons"
     ));
 
+    app->Add(new JChainMultifactoryGeneratorT<GeneratedJets_factory>(
+             "GeneratedJets",
+            {"MCParticles"},
+            {"GeneratedJets"},
+            app
+    ));
+    app->Add(new JChainMultifactoryGeneratorT<ReconstructedJets_factory>(
+            "ReconstructedJets",
+            {"ReconstructedChargedParticles"},
```

Comment:
Please submit separately
```suggestion
            {"ReconstructedParticles"},
```